src/renderers/shared/reconciler/ReactCompositeComponent.js

```diff
@@ -122,10 +124,18 @@ var ReactCompositeComponentMixin = {
    * @final
    * @internal
    */
-  mountComponent: function(rootID, transaction, context) {
+  mountComponent: function(
+    rootID,
+    transaction,
+    nativeParent,
+    nativeContainerInfo,
+    context
+  ) {
     this._context = context;
     this._mountOrder = nextMountID++;
     this._rootNodeID = rootID;
+    this._nativeParent = nativeParent;
+    this._nativeContainerInfo = nativeContainerInfo;
```

Comment:
I don't understand why it wouldn't work unless I misunderstood what _nativeParent does. nativeContainerInfo would still be on the ReactDOMComponent.

I actually think that we could walk back to the nativeParent and reconcile from there so that we can support frags/nulls. Which is what sparked this line of thinking.

Are you saying we would have to crawl back all the way to the root?


> On Oct 12, 2015, at 8:50 PM, Ben Alpert <notifications@github.com> wrote:
> 
> In src/renderers/shared/reconciler/ReactCompositeComponent.js:
> 
> >      this._context = context;
> >      this._mountOrder = nextMountID++;
> >      this._rootNodeID = rootID;
> > +    this._nativeParent = nativeParent;
> > +    this._nativeContainerInfo = nativeContainerInfo;
> I realized this isn't necessary only at the root. Your way doesn't work if we don't store this here because _nativeParent is also a ReactDOMComponent. We could crawl up the tree every time but that feels more wasteful. Essentially, this is where we store the info that doesn't change within the tree but that we need at every level.
> 
> —
> Reply to this email directly or view it on GitHub.
>